packages/core/src/WorkflowExecutionMetadata.ts

```diff
@@ -17,16 +29,30 @@ export function setWorkflowExecutionMetadata(
 	) {
 		return;
 	}
-	executionData.resultData.metadata[String(key).slice(0, 50)] = String(value).slice(0, 255);
+	if (typeof key !== 'string') {
+		throw new WorkflowMetadataValidationError('key', key);
+	}
+	if (typeof value !== 'string') {
+		throw new WorkflowMetadataValidationError('value', key);
+	}
+	executionData.resultData.metadata[key.slice(0, 50)] = value.slice(0, 255);
```

Comment:
Can we also log an error, saying that the contents have been trimmed to fit the limits (and mention the limits)?